// angular-popper/dist/angular-popper.component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var popper_js_1 = require("popper.js");
var PopperComponent = (function () {
    function PopperComponent(el, zone) {
        this.el = el;
        this.zone = zone;
        this.show = false;
        this.closeButton = true;
        this.placement = 'bottom';
        this.close = new core_1.EventEmitter();
    }
    PopperComponent.prototype.ngAfterViewInit = function () {
        this.create();
    };
    PopperComponent.prototype.ngOnChanges = function (changes) {
        if (changes.target && !changes.target.firstChange ||
            changes.placement && !changes.placement.firstChange) {
            this.destroy();
            this.create();
        }
    };
    PopperComponent.prototype.ngOnDestroy = function () {
        this.destroy();
    };
    PopperComponent.prototype.onClose = function () {
        this.show = false;
        this.close.emit();
    };
    PopperComponent.prototype.create = function () {
        var _this = this;
        this.zone.runOutsideAngular(function () {
            _this.popper = new popper_js_1.default(_this.getTargetNode(), _this.el.nativeElement.querySelector('.angular-popper'), {
                placement: _this.placement
            });
        });
    };
    PopperComponent.prototype.destroy = function () {
        var _this = this;
        if (this.popper) {
            this.zone.runOutsideAngular(function () {
                _this.popper.destroy();
            });
            this.popper = null;
        }
    };
    PopperComponent.prototype.getTargetNode = function () {
        if (this.target) {
            if (typeof this.target === 'string') {
                return document.querySelector(this.target);
            }
            else {
                return this.target;
            }
        }
        else {
            return this.el.nativeElement;
        }
    };
    return PopperComponent;
}());
PopperComponent.decorators = [
    { type: core_1.Component, args: [{
                selector: 'angular-popper',
                templateUrl: 'angular-popper.component.html',
                styleUrls: ['./angular-popper.css'],
                changeDetection: core_1.ChangeDetectionStrategy.OnPush
            },] },
];
/** @nocollapse */
PopperComponent.ctorParameters = function () { return [
    { type: core_1.ElementRef, },
    { type: core_1.NgZone, },
]; };
PopperComponent.propDecorators = {
    'show': [{ type: core_1.Input },],
    'closeButton': [{ type: core_1.Input },],
    'placement': [{ type: core_1.Input },],
    'target': [{ type: core_1.Input },],
    'close': [{ type: core_1.Output },],
};
exports.PopperComponent = PopperComponent;
