// angular-popper/dist/angular-popper.bundle.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('popper.js')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/common', 'popper.js'], factory) :
	(factory((global['angular-icon'] = {}),global._angular_core,global._angular_common,global.Popper));
}(this, (function (exports,_angular_core,_angular_common,Popper) { 'use strict';

Popper = Popper && 'default' in Popper ? Popper['default'] : Popper;

var __decorate$1 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
exports.PopperComponent = (function () {
    function PopperComponent(el, zone) {
        this.el = el;
        this.zone = zone;
        this.show = false;
        this.closeButton = true;
        this.placement = 'bottom';
        this.close = new _angular_core.EventEmitter();
    }
    PopperComponent.prototype.ngAfterViewInit = function () {
        this.create();
    };
    PopperComponent.prototype.ngOnChanges = function (changes) {
        if (changes.target && !changes.target.firstChange ||
            changes.placement && !changes.placement.firstChange) {
            this.destroy();
            this.create();
        }
    };
    PopperComponent.prototype.ngOnDestroy = function () {
        this.destroy();
    };
    PopperComponent.prototype.onClose = function () {
        this.show = false;
        this.close.emit();
    };
    PopperComponent.prototype.create = function () {
        var _this = this;
        this.zone.runOutsideAngular(function () {
            _this.popper = new Popper(_this.getTargetNode(), _this.el.nativeElement.querySelector('.angular-popper'), {
                placement: _this.placement
            });
        });
    };
    PopperComponent.prototype.destroy = function () {
        var _this = this;
        if (this.popper) {
            this.zone.runOutsideAngular(function () {
                _this.popper.destroy();
            });
            this.popper = null;
        }
    };
    PopperComponent.prototype.getTargetNode = function () {
        if (this.target) {
            if (typeof this.target === 'string') {
                return document.querySelector(this.target);
            }
            else {
                return this.target;
            }
        }
        else {
            return this.el.nativeElement;
        }
    };
    return PopperComponent;
}());
__decorate$1([
    _angular_core.Input(),
    __metadata("design:type", Object)
], exports.PopperComponent.prototype, "show", void 0);
__decorate$1([
    _angular_core.Input(),
    __metadata("design:type", Object)
], exports.PopperComponent.prototype, "closeButton", void 0);
__decorate$1([
    _angular_core.Input(),
    __metadata("design:type", String)
], exports.PopperComponent.prototype, "placement", void 0);
__decorate$1([
    _angular_core.Input(),
    __metadata("design:type", Object)
], exports.PopperComponent.prototype, "target", void 0);
__decorate$1([
    _angular_core.Output(),
    __metadata("design:type", Object)
], exports.PopperComponent.prototype, "close", void 0);
exports.PopperComponent = __decorate$1([
    _angular_core.Component({
        selector: 'angular-popper',
        templateUrl: 'angular-popper.component.html',
        styleUrls: ['./angular-popper.css'],
        changeDetection: _angular_core.ChangeDetectionStrategy.OnPush
    }),
    __metadata("design:paramtypes", [_angular_core.ElementRef, _angular_core.NgZone])
], exports.PopperComponent);

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
exports.NgPopper = (function () {
    function NgPopper() {
    }
    return NgPopper;
}());
exports.NgPopper = __decorate([
    _angular_core.NgModule({
        imports: [
            _angular_common.CommonModule
        ],
        declarations: [
            exports.PopperComponent
        ],
        exports: [
            exports.PopperComponent
        ]
    })
], exports.NgPopper);

Object.defineProperty(exports, '__esModule', { value: true });

})));
